### PR TITLE
[Feature] LocalFileSource support multiple table

### PR DESCRIPTION
**Pull request by ruanwenjun**
### Purpose of this pull request
related issue: #5677

- LocalFileSource use Factory to create source, support new catalogTable API
- LocalFileSource support define multiple table

### Does this PR introduce _any_ user-facing change?
Add new config option, and compatible with old config.

### How was this patch tested?
test with e2e case

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [X] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_to_console.conf [hunk 1]
```diff
@@ -29,6 +29,8 @@ source {
   LocalFile {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_to_console.conf:29]: how to use  multiple table read？

Comment (ruanwenjun) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_to_console.conf:29]: We can use `local_file_source_configs` to set config for multiple table mode, added doc about this.

File: docs/en/connector-v2/source/LocalFile.md [hunk 2]
```diff
@@ -244,11 +245,71 @@ The compress codec of files and the details that supported as the following show
...
+}
+
+```
+
+### Multiple Table
+
 ```hocon
 
 LocalFile {
+  lo
```
Comment (hailin0) [inline docs/en/connector-v2/source/LocalFile.md]: why use lo？

File: docs/en/connector-v2/source/LocalFile.md [hunk 3]
```diff
@@ -244,11 +245,71 @@ The compress codec of files and the details that supported as the following show
...
+    }
+  ]
+}
+
+```
+
+```hocon
+
+LocalFile {
+  local_file_source_configs = [
```
Comment (hailin0) [inline docs/en/connector-v2/source/LocalFile.md]: Duplicate example

File: docs/en/connector-v2/source/LocalFile.md (hunk 2 above)
Comment (ruanwenjun) [reply docs/en/connector-v2/source/LocalFile.md]: Removed this typo

File: docs/en/connector-v2/source/LocalFile.md (hunk 3 above)
Comment (ruanwenjun) [reply docs/en/connector-v2/source/LocalFile.md]: Done

File: docs/en/connector-v2/source/LocalFile.md [hunk 4]
```diff
@@ -270,6 +277,59 @@ LocalFile {
 
 ```
 
+### Multiple Table
+
+```hocon
+
+LocalFile {
+  local_file_source_configs = [
```
Comment (hailin0) [inline docs/en/connector-v2/source/LocalFile.md]: Use `tables` or `tables_config` or `table_list` or `table_list_config`?

Comment (ruanwenjun) [reply docs/en/connector-v2/source/LocalFile.md]: How about `table_configs`?

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**